File: 1589-maximum-sum-obtained-of-any-permutation/1589-maximum-sum-obtained-of-any-permutation.java
class Solution {
    public int maxSumRangeQuery(int[] nums, int[][] requests) {
        int n=nums.length;
        long res=0,mod=(long )1e9+7;
        int arr[] =new int[n];
        for(int i=0;i<requests.length;i++){
            arr[requests[i][0]]+=1;
            if(requests[i][1]+1<n)arr[requests[i][1]+1]--;
        }
        
        for(int i=1;i<n;i++){
            arr[i]+=arr[i-1];
        }
        
        Arrays.sort(nums);
        Arrays.sort(arr);
    
        for(int i=0;i<n;i++){
            res+=(long)arr[i]*nums[i];
        }
        return (int)(res%mod);
        
        
    }
}

File: 1589-maximum-sum-obtained-of-any-permutation/1589-maximum-sum-obtained-of-any-permutation.cpp
class Solution {
public:
    int maxSumRangeQuery(vector<int>& nums, vector<vector<int>>& requests) {
        int n=nums.size();
        vector<int>vect(n+1);
        for(int i=0;i<requests.size();i++){
            vect[requests[i][0]]++;
            vect[requests[i][1]+1]--;
        }
        for(int i=1;i<n;i++)
        {
            vect[i]+=vect[i-1];
        }
        sort(vect.begin(),vect.end()-1);
        sort(nums.begin(),nums.end());
        
        long ans=0,mod=1e9+7;
        for(int i=0;i<n;i++){
            ans = (ans + (long)(nums[i])*(vect[i]));
        }
        return ans%mod;
    }
};